packages/valory/skills/abstract_round_abci/models.py

```diff
@@ -409,7 +424,8 @@ def __init__(
     def setup(self) -> None:
         """Set up the model."""
         self._round_sequence = RoundSequence(self.abci_app_cls)
-        consensus_params = cast(BaseParams, self.context.params).consensus_params
+        # consensus parameters will not be available if the current skill is abstract
+        consensus_params = getattr(self.context.params, "consensus_params", None)
```

Comment:
I thought we remove them?

Reply:
We do not remove the consensus params from the model, but remove them from the configuration, because we can infer the `max_participants` from `all_participants` and create the consensus params in the model using the latter.

Reply:
I know we can infer `max_participants`. But `threshold` should actually match what's defined onchain. So we will need to pass it in via config.

Reply:
Ok, I see what you mean. So we do not even need the `ConsensusParams` class. The `consensus_threshold` property will become a configured parameter (`setup` param would make the most sense I believe), which will be overridden by the CLI, and retrieved from the on-chain protocol.